docs/design/README.md

```diff
@@ -0,0 +1,552 @@
+        │   ├── 719d1eef5ce4d194d4c5c51171dce8987cb21e88338f1d5dff7554dacabf1004-other.sqlite.bz2
+        │   ├── 7cbab1a2f05c1eda86d3d125497af680d9100665e3eb30c7560926eabbc2c655-primary.xml.gz
+        │   ├── d61fe62ca166eab1cbd502cf21cd977798ef348de49fb35e38a93ebf696eb0a5-filelists.sqlite.bz2
+        │   ├── fcd4dadc59c6d6512410b103a38762bf266389e01a60add12319191c4cc16ec4-other.xml.gz
+        │   └── repomd.xml
+        ├── bottlerocket-x86_64-acpid-2.0.34-1.x86_64.rpm
+        ├── bottlerocket-x86_64-apiclient-0.0-0.x86_64.rpm
+        ├── ...
+        └── bottlerocket-x86_64-wicked-0.6.68-1.x86_64.rpm
+```
+
+The file at `/local/etc/yum.repos.d` can be placed at `/etc/yum.repos.d` later when multiple kits are being aggregated.
+This will alow `dnf` to treat each kit as a yum repository.
+When aggregating kits, the yum repos will be prioritized according to the maintainer's requirements.
+
+The presence of `bootconfig` is questionable.
+These files may need to be moved to the SDK or somewhere else.
+
+The presence of the `migrations` directory is temporary and controversial.
+Only the core project can have migrations, and we expect the migration system to go away.
+Ideally the presence of these will be completely eliminated as OOTB reaches maturity.
+
+### Kit Dependencies
+
+One kit may depend on a package from another kit.
+For example, a maintainer's out-of-tree package may depend on `glibc` from `bottlerocket-core-kit`.
+These dependencies need to be specified at the kit level so that tooling understands which kits are required for a build.
+
+This will be solved by creating a metadata container image where these dependencies are enumerated in a JSON file.
+Consider a kit named `my-awesome-kit`.
+This may be published to a container registry and tagged with versions by the publisher:
+
+```
+registry.com/my-awesome-kit:v0.1.0
+registry.com/my-awesome-kit:v0.2.0
+etc.
+```
+
+Consider that a package within this kit depends on Bottlerocket's `glibc`, and that the required version has changed between this kit's `v0.1.0` and `v0.2.0` versions.
+Our JSON metadata for `my-awsome-kit` will let us know:
+
+```json
+{
+  "kit": {
+    "name": "my-core-kit",
+    "dependencies": {
+      "v0.1.0": [
+        "public.ecr.aws/bottlerocket/bottlerocket-core-kit:v1.15.1"
+      ],
+      "v0.2.0": [
+        "public.ecr.aws/bottlerocket/bottlerocket-core-kit:v1.16.0"
+      ]
+    }
+  }
+}
+```
+
+This metadata can be stored and updated in the kit container registry with a mutable tag so that `twoliter` can easily find it.
+
+```
+registry.com/my-awesome-kit:metadata
+registry.com/my-awesome-kit:v0.1.0
+registry.com/my-awesome-kit:v0.2.0
+etc.
+```
+
+These metadata files will give twoliter the information it needs to resolve, pull, and aggregate all dependency kits needed for a build.
+This dependency resolution may be kept in a lock file inspired by `Cargo.lock` if we find it helpful.
```

Comment:
If we don't have some sort of stable, Git-committable artifact that represents the results of the dependency resolution process, it will create challenges for reproducible builds.